### DISCUSSION TITLE
Makes sense to provide a script to upload a private certificate in APIM?

### DISCUSSION THREAD

**Discussion by mbraekman** [arcus-azure/arcus.scripting#135] [Ideas]
Would it - in order to be complete - make sense to provide an additional function next to the one specified in #134 , to also provide the capability to upload a private certificate in the APIM Certificate store?

Although the actions required are far less complex than uploading a public certificate in the CA Certificate store:
- Get APIM Context
- Create new AzApiManagementCertificate referring to the APIM-context. 
(see: https://docs.microsoft.com/en-us/powershell/module/az.apimanagement/new-azapimanagementcertificate?view=azps-5.5.0)

This would have the same parameters + 1 additional:
- ResourceGroupName
- ServiceName
- CertificateFilePath
- CertificatePassword

It would offer a complete set allowing users to upload both private and public certificate via 1 Arcus-library.

cc: @stijnmoreels , @tomkerkhove

**Comment by stijnmoreels** [comment]
As far to have a (how do they say it) 'whole abstraction' it's indeed something to have both. But I'm not an expert in APIM and certificates used within that context, so I can't bring much added value to the usefulness for this.

**Comment by tomkerkhove** [comment]
Certainly!

**Comment by mbraekman** [comment]
Okay, will be turning this into an "issue" then!
Wanted to avoid that we would be creating functions for the sake of creation functions.

Thanks to both for the feedback!
